form/src/main/java/org/dhis2/form/ui/provider/inputfield/PeriodSelectorProvider.kt

```diff
@@ -30,85 +24,38 @@ fun ProvidePeriodSelector(
     focusRequester: FocusRequester,
     uiEventHandler: (RecyclerViewUiEvents) -> Unit,
 ) {
-    val currentDate = DateUtils.getInstance().getStartOfDay(Date())
-    if ((fieldUiModel.periodSelector?.minDate?.after(currentDate) == true)) {
-        ProvideEmptyPeriodSelector(
-            modifier = modifier,
-            name = fieldUiModel.label,
-            inputStyle = inputStyle,
+    var selectedItem by remember(fieldUiModel.displayName) {
+        mutableStateOf(
+            fieldUiModel.displayName,
         )
-    } else {
-        var selectedItem by remember(fieldUiModel.displayName) {
-            mutableStateOf(
-                fieldUiModel.displayName,
-            )
-        }
-
-        DropdownInputField(
-            modifier = modifier,
-            title = fieldUiModel.label,
-            state = fieldUiModel.inputState(),
-            inputStyle = inputStyle,
-            legendData = fieldUiModel.legend(),
-            supportingTextData = fieldUiModel.supportingText(),
-            isRequiredField = fieldUiModel.mandatory,
-            selectedItem = DropdownItem(selectedItem ?: ""),
-            onResetButtonClicked = {
-                selectedItem = null
-                fieldUiModel.onClear()
-            },
-            onDropdownIconClick = {
-                uiEventHandler(
-                    RecyclerViewUiEvents.SelectPeriod(
-                        uid = fieldUiModel.uid,
-                        title = fieldUiModel.label,
-                        periodType = fieldUiModel.periodSelector!!.type,
-                        minDate = fieldUiModel.periodSelector!!.minDate,
-                        maxDate = fieldUiModel.periodSelector!!.maxDate,
-                    ),
-                )
-            },
-            onFocusChanged = {},
-            focusRequester = focusRequester,
-            expanded = false,
-        )
-    }
-}
-
-@Composable
-fun ProvideEmptyPeriodSelector(
-    modifier: Modifier = Modifier,
-    name: String,
-    inputStyle: InputStyle,
-) {
-    var selectedItem by remember {
-        mutableStateOf("")
     }
 
-    val options = listOf(DropdownItem(stringResource(id = R.string.no_periods)))
-
-    InputDropDown(
+    DropdownInputField(
```

Comment:
Here, are you taking into account that there might not be any periods available? if so we should display an empty option with the string resource "no periods available"

Reply:
Done!